### ISSUE TITLE
ios_framework with multiple objc_library and swift_library deps is broken

### ISSUE THREAD

**Issue by thii** [bazelbuild/rules_apple#1038]
Found another regression since 572aee29236aa6aa850484d5706b79aa2a098219: unable to build an `ios_framework` with multiple `objc_library` and `swift_library` deps. To reproduce:

1. Apply with the following patch:

```diff
diff --git i/test/starlark_tests/targets_under_test/ios/BUILD w/test/starlark_tests/targets_under_test/ios/BUILD
index 3169304..89a1e31 100644
--- i/test/starlark_tests/targets_under_test/ios/BUILD
+++ w/test/starlark_tests/targets_under_test/ios/BUILD
@@ -1107,6 +1107,24 @@ ios_application(
     ],
 )
 
+ios_application(
+    name = "app_with_fmwk_with_multiple_objc_library_and_swift_library_deps",
+    bundle_id = "com.google.example",
+    families = [
+        "iphone",
+        "ipad",
+    ],
+    frameworks = [":fmwk_with_multiple_objc_library_and_swift_library_deps"],
+    infoplists = [
+        "//test/starlark_tests/resources:Info.plist",
+    ],
+    minimum_os_version = "8.0",
+    tags = FIXTURE_TAGS,
+    deps = [
+        "//test/starlark_tests/resources:objc_main_lib",
+    ],
+)
+
 ios_framework(
     name = "fmwk_no_version",
     hdrs = ["//test/starlark_tests/resources:common.h"],
@@ -1347,6 +1365,24 @@ ios_framework(
     ],
 )
 
+ios_framework(
+    name = "fmwk_with_multiple_objc_library_and_swift_library_deps",
+    bundle_id = "com.google.example.framework",
+    families = [
+        "iphone",
+        "ipad",
+    ],
+    infoplists = [
+        "//test/starlark_tests/resources:Info.plist",
+    ],
+    minimum_os_version = "8.0",
+    tags = FIXTURE_TAGS,
+    deps = [
+        "//test/starlark_tests/resources:swift_main_lib",
+        "//test/starlark_tests/resources:objc_shared_lib",
+    ],
+)
+
 # ---------------------------------------------------------------------------------------
 
 objc_library(
```

2. Build with

```
bazel build //test/starlark_tests/targets_under_test/ios:app_with_fmwk_with_multiple_objc_library_and_swift_library_deps
```

```
ERROR: /Users/user/rules_apple/test/starlark_tests/targets_under_test/ios/BUILD:1368:14: in //apple/internal/aspects:swift_dynamic_framework_aspect.bzl%swift_dynamic_framework_aspect aspect on swift_runtime_linkopts rule //test/starlark_tests/targets_under_test/ios:fmwk_with_multiple_objc_library_and_swift_library_deps.swift_runtime_linkopts:
Traceback (most recent call last):
        File "/Users/user/rules_apple/apple/internal/aspects/swift_dynamic_framework_aspect.bzl", line 94, column 13, in _swift_dynamic_framework_aspect_impl
                fail(
Error in fail: error: Found a mix of swift_library and other rule dependencies. Swift dynamic frameworks expect a single swift_library dependency.
ERROR: Analysis of target '//test/starlark_tests/targets_under_test/ios:app_with_fmwk_with_multiple_objc_library_and_swift_library_deps' failed; build aborted: Analysis of target '//test/starlark_tests/targets_under_test/ios:fmwk_with_multiple_objc_library_and_swift_library_deps.swift_runtime_linkopts' failed
INFO: Elapsed time: 6.621s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (10 packages loaded, 201 targets configured)
```

This builds successfully before the changes in 572aee29236aa6aa850484d5706b79aa2a098219.

**Comment by mccorkill1** [comment]
Thank you for finding this! That error needs to stay in place for the ios_dynamic_framework rule, but shouldn't be enforced if it's just an ios_framework. I will start looking into this today.

**Comment by thii** [comment]
Resolved by #1043. Thanks for the fix.
